pennylane/ops/op_math/exp.py

```diff
@@ -362,6 +366,12 @@ def _trotter_decomposition(self, ops: List[Operator], coeffs: List[complex]):
 
         return op_list * self.num_steps  # apply operators ``num_steps`` times
 
+    @property
+    def has_matrix(self):
+        return not (
+            math.get_interface(self.scalar) == "autograd" and math.requires_grad(self.scalar)
+        ) and self.base.has_matrix
+
```

Comment:
I've been locally running the the intro to qaoa and falqon demo's, and I don't think we are actually needing this part.

Reply:
Ah. I understand now.  If the base is a Hamiltonian, it won't have a matrix. So the evolution/ exp wouldn't have a matrix if the base is a Hamiltonian, BUT it's presence in the `observables` list was overriding that behaviour.  We remove `Exp` and `Evolution` from this observables list and everything works fine.
```suggestion
```